test/cpp/util/tls_test_credentials.h

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_TEST_CPP_UTIL_TLS_TEST_CREDENTIALS_H
+#define GRPC_TEST_CPP_UTIL_TLS_TEST_CREDENTIALS_H
+
+#include <grpcpp/security/credentials.h>
+#include <grpcpp/security/server_credentials.h>
+#include <memory>
+#include <thread>
+#include <vector>
+
+using ::grpc_impl::experimental::TlsCredentialReloadConfig;
+using ::grpc_impl::experimental::TlsCredentialsOptions;
+using ::grpc_impl::experimental::TlsKeyMaterialsConfig;
+using ::grpc_impl::experimental::TlsServerAuthorizationCheckConfig;
+
+namespace grpc {
+namespace testing {
+
+struct TlsThread {
+  void Join() {
+    if (thread_started_) {
+      thread_.join();
+      thread_started_ = false;
+    }
+  }
+
+  std::thread thread_;
```

Comment:
Please use `grpc_core::Thread` instead.

Reply:
Just to check: I should still use `grpc_core::Thread` even if this isn't in the C-core? I thought any C++11 API's were ok, at least in the C++ wrapper.

Reply:
Done.